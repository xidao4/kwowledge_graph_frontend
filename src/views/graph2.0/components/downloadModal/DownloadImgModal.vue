<template>
    <a-modal
            title="下载图谱"
            @cancel="cancel"
            :visible="showDownloadImgModal"
    >
        <template slot="footer">
            <a-button type="primary" @click="download">
                下载
            </a-button>
            <a-button @click="cancel">
                取消
            </a-button>
        </template>

        <a-form :form="imgForm" :label-col="{ span: 5 }" :wrapper-col="{ span: 19 }">
            <a-form-item
                    label="文件名"
                    help='默认文件名为"知识图谱"'>
                <a-input style="border-radius: 20px; height: 40px"
                         v-model="name"
                />
            </a-form-item>
            <a-form-item label="无背景">
                <a-switch default-checked @change="changeBackground" />
            </a-form-item>
        </a-form>
    </a-modal>
</template>

<script>
    import { mapGetters, mapMutations } from 'vuex';

    export default {
        name: "DownloadImgModal",
        data(){
            return {
                imgForm: this.$form.createForm(this, { name: 'imgForm' }),
                noBackground: true,
                name: '知识图谱'
            }
        },
        props: {
            type: {
                type: String,
                default: 'png'
            }
        },
        computed:{
            ...mapGetters([
                'currentGraph',
                'showDownloadImgModal'
            ])
        },
        methods: {
            ...mapMutations([
                'set_showDownloadImgModal'
            ]),
            cancel(){
                this.set_showDownloadImgModal(false);
            },
            download(){
                let name = this.name || "知识图谱";
                let type = this.type || "png";
                let paramType = 'image/' + (this.type || "png");
                let imgUrl = '';
                if(this.noBackground){
                    imgUrl = this.currentGraph.toDataURL(paramType);
                } else {
                    imgUrl = this.currentGraph.toDataURL(paramType, '#FFFFFF');
                }
                const imgLink = document.createElement('a');
                imgLink.download = name + '.' + type;
                imgLink.style.display = 'none';
                imgLink.href = imgUrl;
                document.body.appendChild(imgLink);
                imgLink.click();
                URL.revokeObjectURL(imgLink.href);
                document.body.removeChild(imgLink)

                this.set_showDownloadImgModal(false);
            },
            changeBackground(checked) {
                this.noBackground = checked;
            },
        }
    }
</script>

<style scoped>

</style>
