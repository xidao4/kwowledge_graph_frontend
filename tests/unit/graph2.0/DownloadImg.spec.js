import { mount, createLocalVue } from '@vue/test-utils'
import DownloadImgModal from '@/views/graph2.0/components/downloadModal/DownloadImgModal.vue'
import Antd from 'ant-design-vue';
import Vuex from 'vuex';

const localVue = createLocalVue();
localVue.use(Antd);
localVue.use(Vuex);

describe('DownloadImgModal.vue', () => {
    let getters;
    let mutations;
    let store;
    let wrapper;

    beforeEach(() => {
        getters = {
            showDownloadImgModal: () => true,
            currentGraph: () => {
                return {
                    toDataURL: jest.fn()
                }
            }
        };
        mutations = {
            set_showDownloadImgModal: jest.fn()
        };
        store = new Vuex.Store({
            state: {},
            getters,
            mutations
        });
        wrapper = mount(DownloadImgModal, {
            store,
            localVue,
            data() {
                return {
                    name: '知识图谱'
                }
            },
            propsData: {
                type: ''
            }
        });
    });
    it('download snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('用户将图谱名改为空', async () => {
        global.URL.revokeObjectURL = jest.fn();
        let buttons = wrapper.findAll('button');
        for(let i = 0; i < buttons.length; i++){
            let button = buttons.at(i);
            button.trigger('click');
        }
        await wrapper.setData(
            {
                name: '',
                noBackground: true
            }
        );
        for(let i = 0; i < buttons.length; i++){
            let button = buttons.at(i);
            button.trigger('click');
        }
        expect(mutations.set_showDownloadImgModal).toHaveBeenCalled();
    });
    it('指定图片类型', async () => {
        global.URL.revokeObjectURL = jest.fn();
        await wrapper.setProps(
            {
                type: 'jpg'
            }
        );
        let buttons = wrapper.findAll('button');
        for(let i = 0; i < buttons.length; i++){
            let button = buttons.at(i);
            button.trigger('click');
        }
        expect(mutations.set_showDownloadImgModal).toHaveBeenCalled();
    });
});
