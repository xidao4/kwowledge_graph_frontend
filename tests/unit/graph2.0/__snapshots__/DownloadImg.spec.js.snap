// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadImgModal.vue download snapshot 1`] = ``;
